src/Symfony/Bridge/Doctrine/ManagerRegistry.php

```diff
@@ -45,31 +45,60 @@ protected function resetService($name): void
 
             return;
         }
-        if (!$manager instanceof LazyLoadingInterface) {
-            throw new \LogicException(\sprintf('Resetting a non-lazy manager service is not supported. Declare the "%s" service as lazy.', $name));
+        if (\PHP_VERSION_ID < 80400) {
+            if (!$manager instanceof LazyLoadingInterface) {
+                throw new \LogicException(\sprintf('Resetting a non-lazy manager service is not supported. Declare the "%s" service as lazy.', $name));
+            }
+            trigger_deprecation('symfony/doctrine-bridge', '7.3', 'Support for proxy-manager is deprecated.');
```

Comment:
This case should never happen (who would use proxy-manager directly instead of Symfony's lazy services for the Doctrine EM!?), still added for completeness.

Reply:
isn't this deprecation always triggered on PHP 8.3 and older ?

Reply:
it's guarded by `if (!$manager instanceof LazyLoadingInterface) {` (note the `!`) so only when ppl use proxy-manager, which we don't anymore thanks to var-exporter.

Reply:
oh, I missed that the case of lazy objects using var-exporter are handled _before_ the check for PHP 8.4, so this is not the code path we reach in modern projects.